sky/spot/spot_utils.py

```diff
@@ -260,12 +260,17 @@ def stream_logs_by_id(job_id: int, follow: bool = True) -> str:
             job_statuses = backend.get_job_status(handle, stream_logs=False)
             job_status = list(job_statuses.values())[0]
             assert job_status is not None, 'No job found.'
-            logger.info(f'Logs finished for job {job_id} '
-                        f'(status: {job_status.value}).')
-            break
-        logger.info(
-            f'INFO: (Log streaming) Got return code {returncode}. Retrying '
-            f'in {JOB_STATUS_CHECK_GAP_SECONDS} seconds.')
+            if job_status != job_lib.JobStatus.CANCELLED:
+                logger.info(f'Logs finished for job {job_id} '
+                            f'(status: {job_status.value}).')
+                break
+            logger.info('INFO: (Log streaming) The job is cancelled. Waiting '
+                        'for the controller process to update the status in '
+                        f'{JOB_STATUS_CHECK_GAP_SECONDS} seconds.')
```

Comment:
Nit: to me (a user) this seems a bit hard to understand (and does the user need to know this?).

Reply:
Updated it a bit. Since we cannot distinguish the job being cancelled by controller due to partially preemption and actual user's cancellation before the spot status being updated, we have to wait a while until the spot status is updated.